src/java.base/share/classes/java/lang/constant/ClassDesc.java

```diff
@@ -167,14 +168,27 @@ default ClassDesc arrayType() {
      *
      * @param rank the rank of the array
      * @return a {@linkplain ClassDesc} describing the array type
-     * @throws IllegalArgumentException if the rank is less than or equal to zero or if the rank of the resulting array type is
+     * @throws IllegalArgumentException if the rank is less than or
+     * equal to zero or if the rank of the resulting array type is
      * greater than 255
      * @jvms 4.4.1 The CONSTANT_Class_info Structure
      */
     default ClassDesc arrayType(int rank) {
-        int currentDepth = ConstantUtils.arrayDepth(descriptorString());
-        if (rank <= 0 || currentDepth + rank > ConstantUtils.MAX_ARRAY_TYPE_DESC_DIMENSIONS)
-            throw new IllegalArgumentException("rank: " + currentDepth + rank);
+        int netRank;
+        if (rank <= 0) {
+            throw new IllegalArgumentException("rank " + rank + "is not a positive value");
+        }
+        try {
+            int currentDepth = ConstantUtils.arrayDepth(descriptorString());
+            netRank = Math.addExact(currentDepth, rank);
+            if (netRank > ConstantUtils.MAX_ARRAY_TYPE_DESC_DIMENSIONS) {
+                throw new IllegalArgumentException("rank: " + netRank +
```

Comment:
This doesn’t need to use <code>[Math.addExact]\(<var>currentDepth</var>, <var>rank</var>\)</code>, as <code><var>currentDepth</var></code> will always be in [0, 255], and <code><var>rank</var></code> here is confined to [1, 2<sup>31</sup>-1], so it’s possible to check for <code><var>netRank</var> &lt; 0</code> and use <code>[Integer.toUnsignedString]\(<var>netRank</var>\)</code> (or <code>[Integer.toUnsignedLong]\(<var>netRank</var>\)</code> and let <code>[StringConcatFactory]</code> deal with it):

```suggestion
            netRank = currentDepth + rank;
            if (netRank > ConstantUtils.MAX_ARRAY_TYPE_DESC_DIMENSIONS || netRank < 0) {
                throw new IllegalArgumentException("rank: " + Integer.toUnsignedLong(netRank) +
```

[Integer.toUnsignedLong]: https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Integer.html#toUnsignedLong(int)
[Integer.toUnsignedString]: https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Integer.html#toUnsignedString(int)
[Math.addExact]: https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Math.html#addExact(int,int)
[StringConcatFactory]: https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/invoke/StringConcatFactory.html

Reply:
Notice the overflow case is already handled by the `ArithmeticException` caught in the larger try block.